File: 2021/21-2.java
package net.codejava;

import java.io.File;
import java.io.FileNotFoundException;  // Import this class to handle errors
import java.util.Scanner; // Import the Scanner class to read text files

public class AOC {

	public static void main(String[] args) {		
		int x = 0; int y = 0; int y2 = 0; int aim = 0;
		try 
		{
			File myObj = new File("21-2.txt");
			Scanner myReader = new Scanner(myObj);
			
		    while (myReader.hasNextLine()) 
		    {
		          String data[] = myReader.nextLine().split(" ");
		          if(data[0].indexOf("forward") != -1)
		          {
		        	  x += Integer.valueOf(data[1]);
		          	  y2 += aim * Integer.valueOf(data[1]);
		          }
		          if(data[0].indexOf("down") != -1)
		          {
		        	  y += Integer.valueOf(data[1]);
		          	  aim += Integer.valueOf(data[1]);
		          }
		          if(data[0].indexOf("up") != -1)
		          {
		        	  y -= Integer.valueOf(data[1]);
		          	  aim -= Integer.valueOf(data[1]);
		          }
		    }
		    System.out.println("part 1 - " + x*y);
		    System.out.println("part 2 - " + x*y2);
		}		
		catch (FileNotFoundException e) 
		{
		      System.out.println("An error occurred.");
		      e.printStackTrace();
		}
	}
}

File: 2021/21-2.py
r,x,y = [i.split() for i in open('21-2.txt').read().split('\n')],0,0
for i in r:
    if i[0] == 'forward': x += int(i[1])
    if i[0] == 'down': y += int(i[1])
    if i[0] == 'up': y -= int(i[1])    
print('part 1 -',x*y)
x,y,aim = 0,0,0
for i in r:
    if i[0] == 'forward':
        x += int(i[1])
        y += aim * int(i[1]) 
    if i[0] == 'down': aim += int(i[1])
    if i[0] == 'up': aim -= int(i[1]) 
print('part 2 -',x*y)
